pkg/nsx/services/securitypolicy/store.go

```diff
@@ -114,35 +114,36 @@ func (securityPolicyStore *SecurityPolicyStore) Apply(i interface{}) error {
 }
 
 func (securityPolicyStore *SecurityPolicyStore) GetByKey(key string) *model.SecurityPolicy {
-	var securityPolicy model.SecurityPolicy
+	var securityPolicy *model.SecurityPolicy
 	obj := securityPolicyStore.ResourceStore.GetByKey(key)
 	if obj != nil {
-		securityPolicy = obj.(model.SecurityPolicy)
+		securityPolicy = obj.(*model.SecurityPolicy)
 	}
-	return &securityPolicy
+	return securityPolicy
 }
 
-func (securityPolicyStore *SecurityPolicyStore) GetByIndex(key string, value string) []model.SecurityPolicy {
-	securityPolicies := make([]model.SecurityPolicy, 0)
+func (securityPolicyStore *SecurityPolicyStore) GetByIndex(key string, value string) []*model.SecurityPolicy {
+	securityPolicies := make([]*model.SecurityPolicy, 0)
 	objs := securityPolicyStore.ResourceStore.GetByIndex(key, value)
 	for _, securityPolicy := range objs {
-		securityPolicies = append(securityPolicies, securityPolicy.(model.SecurityPolicy))
+		securityPolicies = append(securityPolicies, securityPolicy.(*model.SecurityPolicy))
 	}
 	return securityPolicies
 }
 
 func (ruleStore *RuleStore) Apply(i interface{}) error {
 	sp := i.(*model.SecurityPolicy)
 	for _, rule := range sp.Rules {
+		tempRule := rule
```

Comment:
why we always need a temp var: tempRule when iterating sp.rules? 
we see this is happening for other Apply functions as well.

Reply:
to remove warning by lint

Reply:
what lint do you use? golang-ci-lint?